### DISCUSSION TITLE
what do I need to know about wallet connect authorizations?

### DISCUSSION THREAD

**Discussion by fanboyn** [WalletConnect/walletconnect-monorepo#3701] [Developer Support]
I found that they are using a relay of WalletConnect and passing uri of their own verify service for pairing.

and I didn't find out how verify service should work. is there some example?

is there some deal between WalletConnect and World ID or i can easily implement the same feature as in World ID? (to connect dapp to the mobile app to pass auth data/proof by this connection)

**Comment by horacio3m** [comment]
User Interaction: When a user wants to connect their mobile wallet to a DApp, the DApp generates a QR code or a deep link that contains a unique session request.

Scanning QR Code or Clicking Link: The user scans the QR code or clicks the deep link using their mobile wallet app.

Authorization: The wallet app displays the details of the connection request, such as the DApp's name and permissions it's requesting. The user can then approve or reject the request.

Session Establishment: If the user approves the request, a secure communication session is established between the DApp and the wallet.
Communication: The DApp and wallet can now communicate securely. This enables actions like signing transactions or verifying user identity.

Regarding the "verify service," it's possible that a specific DApp or project has implemented its own verification service on top of WalletConnect for additional security or identity verification purposes. The specifics of how this verification service works would depend on the implementation of that particular project. You may need to refer to the project's documentation or reach out to its developers for more information on how their verify service operates.
